packages/typescript-estree/tests/lib/convert.test.ts

```diff
@@ -252,13 +232,62 @@ describe('convert', () => {
     for (const code of jsDocCode) {
       const ast = convertCode(code);
 
-      const instance = new Converter(ast, {
-        errorOnUnknownASTType: false,
-        shouldPreserveNodeMaps: false,
-      });
+      const instance = new Converter(ast);
       expect(() => instance.convertProgram()).toThrow(
         'JSDoc types can only be used inside documentation comments.',
       );
     }
   });
+
+  describe('errorOnInvalidAST', () => {
+    function generateTest(title: string, code: string, error: string): void {
+      it(`does not throw an error for ${title} when errorOnInvalidAST is false`, () => {
+        const ast = convertCode(code);
+
+        const instance = new Converter(ast);
+
+        expect(() => instance.convertProgram()).not.toThrow();
+      });
+
+      it(`throws an error for ${title} when errorOnInvalidAST is true`, () => {
+        const ast = convertCode(code);
+
+        const instance = new Converter(ast, {
+          errorOnInvalidAST: true,
+        });
+
+        expect(() => instance.convertProgram()).toThrow(error);
+      });
+    }
+
+    generateTest(
+      'a decorator on an enum declaration',
+      '@decl let value;',
+      'Decorators are not valid here.',
+    );
+
+    generateTest(
+      'a decorator on an interface declaration',
+      '@decl interface _ {};',
+      'Decorators are not valid here.',
+    );
+
+    generateTest(
+      'a decorator on a type alias declaration',
+      '@decl type _ = {};',
+      'Decorators are not valid here.',
+    );
+
+    generateTest(
+      'a decorator on a variable statement',
+      '@decl let value;',
+      'Decorators are not valid here.',
+    );
+
+    generateTest(
+      'a variable declaration with no variables',
+      'const;',
+      'A variable declaration list must have at least one variable declarator.',
+    );
```

Comment:
These should use the testing infrastructure add in https://github.com/typescript-eslint/typescript-eslint/pull/3258/files#diff-f9c12fa5e1d8a19e99f46b09e5133914cb8a51270e162893e4603e90567f2af9. It was just easier to quickly iterate here. 😄